src/magicbox/InputManager.ts

```diff
@@ -261,6 +267,18 @@ export class InputManager {
     }
   }
 
+  private handleLeftRightArrow(e: KeyboardEvent) {
+    if ($$(this.root).find(`.${Component.computeCssClassNameForType('QuerySuggestPreview')}`)) {
+      if (this.onkeydown == null || this.onkeyup(e.keyCode || e.which)) {
+        this.onInputChange();
+      } else {
+        e.preventDefault();
+      }
+    } else {
```

Comment:
I would prefer to flip the order to reduce the nesting of the `if/else` and remove the `else` statements. I find it makes it much easier to follow the logic. e.g.

```
private handleLeftRightArrow(e: KeyboardEvent) {
    const querySuggestPreview = (this.root).find(`.${Component.computeCssClassNameForType('QuerySuggestPreview')}`);

    if (!querySuggestPreview) {
        return this.onchangecursor(); 
    }

    const inputChanged = this.onkeydown == null || this.onkeyup(e.keyCode || e.which) // even better would be to split the condition into two variables. I don't fully understand the two parts of the ||
    inputChanged ? this.onInputChange() : e.preventDefault();
}
```